### PR TITLE
Porting to pika 0.7.0

### PR DESCRIPTION
**Pull request by aurianer**

### CODE REVIEW

**Comment by aurianer** [comment]
bors try

**Comment by msimberg** [comment]
bors try-

**Comment by msimberg** [comment]
bors try

**Comment by aurianer** [comment]
bors try

**Comment by aurianer** [comment]
bors try

**Comment by aurianer** [comment]
bors try-

**Comment by aurianer** [comment]
bors try

**Comment by msimberg** [comment]
@aurianer note that the original build failed because of kubernetes problems. I restarted the failed builds and this looks ok now.

File: spack/packages/dla-future/package.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("umpire@6:")
 
     depends_on("pika +mpi")
-    depends_on("pika@0.6:")
+    depends_on("pika@main")
```
Comment (msimberg) [inline spack/packages/dla-future/package.py:49]: To do before merging:
```suggestion
    depends_on("pika@0.7:")
```

Also, update the spack hash in the CI configuration to get pika 0.7.0.

Comment (aurianer) [reply spack/packages/dla-future/package.py:49]: Yes it's planned but my PR in spack is not merged yet, will do right after

Comment (msimberg) [reply spack/packages/dla-future/package.py]: Spack hash still needs updating.

Comment (aurianer) [reply spack/packages/dla-future/package.py]: Sorry for the delay :/

**Comment by aurianer** [comment]
bors try

**Comment by aurianer** [comment]
bors try

**Review by msimberg — APPROVED** [review]

**Review by rasolca — APPROVED** [review]

**Pull request merged**